storage/innobase/row/row0import.cc

```diff
@@ -1321,28 +1321,20 @@ row_import::match_schema(
 	THD*		thd) UNIV_NOTHROW
 {
 	/* Do some simple checks. */
+	const unsigned relevant_flags = m_flags & ~DICT_TF_MASK_DATA_DIR;
+	const unsigned relevant_table_flags =
+	    m_table->flags & ~DICT_TF_MASK_DATA_DIR;
 
-	if (m_flags != m_table->flags) {
-		if (dict_tf_to_row_format_string(m_flags) !=
-				dict_tf_to_row_format_string(m_table->flags)) {
+	if (relevant_flags != relevant_table_flags) {
```

Comment:
I guess the original patch never updated the FSP_MASK_FLAG and DICT_TF_FLAG on import?
FSP flag is present on the disk.
DICT_TF flag on dictionary.

After a restart, do we get a flag mismatch error or warning?

Reply:
Hello @satya-bodapati , I tested show create table, SELECT/INSERT after  restart of server. It is not reporting any error or warning on the table.

Reply:
I debugged a bit and found out that DATA_DIR flag is not used for flag comparison

` /* Validate this single-table-tablespace with the data dictionary,
        but do not compare the DATA_DIR flag, in case the tablespace was
        remotely located. *
`

Reply:
Datafile::validate_to_dd()